playwright/src/main/java/com/microsoft/playwright/impl/LocatorImpl.java

```diff
@@ -67,7 +67,8 @@ public LocatorImpl(FrameImpl frame, String selector, LocatorOptions options) {
       if (options.hasText != null) {
         if (options.hasText instanceof Pattern) {
           Pattern pattern = (Pattern) options.hasText;
-          selector += " >> :scope:text-matches(" + escapeWithQuotes(pattern.pattern()) + ", \"" + toJsRegexFlags(pattern) + "\")";
+          String jsRegex = "/" + pattern.pattern() + "/" + toJsRegexFlags(pattern);
+          selector += " >> has=" + gson().toJson("text=" + jsRegex);
```

Comment:
Does this match the others? I think the gson encoding should just be around jsRegex. Might not matter, but match whichever JS does

Reply:
yes same as others.